src/main/scala/fixedpoint/FixedPoint.scala

```diff
@@ -428,4 +429,22 @@ sealed class FixedPoint private[fixedpoint] (width: Width, private var _inferred
         s"FixedPoint$width$binaryPoint$suffix"
     }
   }
+
+  def floor(): FixedPoint = {
+    // Set the fractional part to zeroes
+    val floored = Cat(data(data.widthOption.get - 1, binaryPoint.get), 0.U((binaryPoint.get).W)).asSInt
+    FixedPoint.fromData(binaryPoint, floored)
+  }
+
+  def ceil(): FixedPoint = {
+    // Get a number with the fractional part set to ones
+    val almost_one = 0.U(data.getWidth.W) + ~(0.U(this.binaryPoint.get.W))
+    // Add it to the number and floor it
+    (this + FixedPoint.fromData(binaryPoint, 0.S(data.getWidth.W) + almost_one.asSInt)).floor()
```

Comment:
I think we can completely remove this part:
```scala
0.S(data.getWidth.W)
```
...and just derive the fixed-point value from `almostOne.asSInt`.